src/builder/scripts/SwapBuilderScripts.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.27;
+
+import {IQuarkWallet} from "quark-core/src/interfaces/IQuarkWallet.sol";
+import {Actions} from "src/builder/Actions.sol";
+import {Accounts} from "src/builder/Accounts.sol";
+import {BridgeRoutes} from "src/builder/BridgeRoutes.sol";
+import {EIP712Helper} from "src/builder/EIP712Helper.sol";
+import {Math} from "src/lib/Math.sol";
+import {MorphoInfo} from "src/builder/MorphoInfo.sol";
+import {Strings} from "src/builder/Strings.sol";
+import {PaycallWrapper} from "src/builder/PaycallWrapper.sol";
+import {QuotecallWrapper} from "src/builder/QuotecallWrapper.sol";
+import {PaymentInfo} from "src/builder/PaymentInfo.sol";
+import {TokenWrapper} from "src/builder/TokenWrapper.sol";
+import {QuarkOperationHelper} from "src/builder/QuarkOperationHelper.sol";
+import {List} from "src/builder/List.sol";
+import {QuarkBuilderBase} from "src/builder/QuarkBuilderBase.sol";
+
+contract SwapBuilderScripts is QuarkBuilderBase {
+    struct ZeroExSwapIntent {
+        uint256 chainId;
+        address entryPoint;
+        bytes swapData;
+        address sellToken;
+        uint256 sellAmount;
+        address buyToken;
+        uint256 buyAmount;
+        address feeToken;
+        uint256 feeAmount;
+        address sender;
+        bool isExactOut;
+        uint256 blockTimestamp;
+    }
+
+    function swap(
+        ZeroExSwapIntent memory swapIntent,
+        Accounts.ChainAccounts[] memory chainAccountsList,
+        PaymentInfo.Payment memory payment
+    ) external pure returns (BuilderResult memory) {
+        // If the action is paid for with tokens, filter out any chain accounts that do not have corresponding payment information
+        if (payment.isToken) {
+            chainAccountsList = Accounts.findChainAccountsWithPaymentInfo(chainAccountsList, payment);
+        }
+
+        IQuarkWallet.QuarkOperation memory operation;
+        Actions.Action memory action;
+        IQuarkWallet.QuarkOperation[] memory quarkOperationsArray;
+        Actions.Action[] memory actionsArray;
+
+        {
```

Comment:
Again, might be good to comment here that this scope is introduced to avoid `"stack too deep"`